### ISSUE TITLE
Msking for BatchNorm

### ISSUE THREAD

**Issue by hadaev8** [pytorch/pytorch#30414]
NLP tasks are very popular.
For such tasks, we usually need to process training samples of different lengths.
To create a batch of the same length, we have to add zeros, which interferes batchnorm and decrease performance of the model.

**Comment by izdeby** [comment]
hi, @hadaev8, we are actually working on adding nested tensor that will solve this issue. As of right now it is in active development. If you would like to check out experimental version (use at your own risk), please, check out [this](https://github.com/pytorch/nestedtensor).

For now i will close the issue.
